### ISSUE TITLE
performance: `notify-references` could split linting into dependents and dependencies

### ISSUE THREAD

**Issue by mainej** [clojure-lsp/clojure-lsp#1028]
**Describe the bug**
When notifying references that a file ha changed, dependents should respond to definition changes, and so need clj-kondo's unresolved-var and invalid-arity lints. Dependencies should respond to usage changes, and so need clojure-lsp's unused-public-var lint. It might be more efficient to process the two separately, asking for different kinds of linting of each. We'd have to confirm that this actually helps performance.

**Comment by ericdallo** [comment]
Yes, not sure what we should do in code exactly, but sounds like a good way to distinguish linting

**Comment by mainej** [comment]
`file-management/reference-filenames` currently calculates separate sets of dependent and dependency files, then merges them together. But instead of merging them, we could run different jobs on each set. For the dependency files, we could calculate, save and publish their unused-public-var lint, bypassing kondo. For the dependent files, we could send them through kondo, but without a custom-lint-fn.

**Comment by ericdallo** [comment]
I see now, sounds like a good move indeed
